tests/core/test_geometric_sources.py

```diff
@@ -45,3 +47,62 @@ def test_multiple_lines_source():
         algo.points = points[:, :1]
     with pytest.raises(ValueError, match='>=2 points need to define multiple lines.'):
         algo.points = points[0, :]
+
+
+@pytest.fixture()
+def bunny():
+    return examples.download_bunny()
+
+
+@pytest.mark.parametrize("is_negative", (True, False))
+@pytest.mark.parametrize("delta", ([0, 0, 0], [1e-8, 0, 0], [0, 0, 1e-8]))
+def test_translate_direction_collinear(is_negative, delta, bunny):
+    mesh_in = bunny
+    direction = np.array([0.0, 1.0, 0.0]) + delta
+    if is_negative:
+        direction *= -1
+    mesh_out = mesh_in.copy()
+    translate(mesh_out, direction=direction)
+    points_in = mesh_in.points
+    points_out = mesh_out.points
+
+    if is_negative:
+        assert np.allclose(points_in[:, 0], -points_out[:, 1])
+        assert np.allclose(points_in[:, 1], points_out[:, 0])
+        assert np.allclose(points_in[:, 2], points_out[:, 2])
+    else:
+        assert np.allclose(points_in[:, 0], points_out[:, 1])
+        assert np.allclose(points_in[:, 1], -points_out[:, 0])
+        assert np.allclose(points_in[:, 2], points_out[:, 2])
+
+
+def test_translate_rotations():
```

Comment:
Note that the above lines in this module were just copied over from `test_geometric_object.py`  The subsequent lines are new specific to the changes in this PR.

Reply:
I created #4978 to split PR about it. Let's merge `main` into this PR later to make the differences easier to understand.